src/snowflake/snowpark/dataframe_reader.py

```diff
@@ -1024,3 +1073,480 @@ def _read_semi_structured_file(self, path: str, format: str) -> DataFrame:
+                struct_schema = type_string_to_type_object(custom_schema)
+                if not isinstance(struct_schema, StructType):
+                    logger.error(f"Invalid schema string: {custom_schema}")
+                    raise ValueError(
+                        f"Invalid schema string: {custom_schema}. "
+                        f"You should provide a valid schema string representing a struct type."
+                    )
+            elif isinstance(custom_schema, StructType):
+                struct_schema = custom_schema
+            else:
+                logger.error(f"Invalid schema type: {type(custom_schema)}")
+                raise TypeError(f"Invalid schema type: {type(custom_schema)}.")
+
+        select_query = generate_select_query(
+            table, struct_schema, dbms_type, driver_info
+        )
+        logger.debug(f"Generated select query: {select_query}")
+        if column is None:
+            if (
+                lower_bound is not None
+                or upper_bound is not None
+                or num_partitions is not None
+            ):
+                logger.error("column is None but bounds or partitions are specified")
+                raise ValueError(
+                    "when column is not specified, lower_bound, upper_bound, num_partitions are expected to be None"
+                )
+            if predicates is None:
+                partitioned_queries = [select_query]
+            else:
+                partitioned_queries = generate_sql_with_predicates(
+                    select_query, predicates
+                )
+        else:
+            if lower_bound is None or upper_bound is None or num_partitions is None:
+                logger.error("column is specified but bounds or partitions are missing")
+                raise ValueError(
+                    "when column is specified, lower_bound, upper_bound, num_partitions must be specified"
+                )
+
+            column_type = None
+            for field in struct_schema.fields:
+                if field.name.lower() == column.lower():
+                    column_type = field.datatype
+            if column_type is None:
+                logger.error("Specified column does not exist in schema")
+                raise ValueError("Column does not exist")
+
+            if not isinstance(column_type, _NumericType) and not isinstance(
+                column_type, DateType
+            ):
+                logger.error(f"Unsupported column type: {column_type}")
+                raise ValueError(
+                    f"unsupported type {column_type}, we support numeric type and date type"
+                )
+            partitioned_queries = self._generate_partition(
+                select_query,
+                column_type,
+                column,
+                lower_bound,
+                upper_bound,
+                num_partitions,
+            )
+        with tempfile.TemporaryDirectory() as tmp_dir:
```

Comment:
Where is this temp directory created in a SP/notebook?

Reply:
it's still under /tmp dir
<img width="443" alt="Screenshot 2025-02-28 at 11 52 58 AM" src="https://github.com/user-attachments/assets/9bff4bcd-eb25-49ca-b421-9ad31db96b83" />